arch/risc-v/src/mpfs/mpfs_serial.c

```diff
@@ -1343,6 +1347,47 @@ static bool up_rxavailable(struct uart_dev_s *dev)
   return ((up_serialin(priv, MPFS_UART_LSR_OFFSET) & UART_LSR_DR) != 0);
 }
 
+#ifdef CONFIG_SERIAL_IFLOWCONTROL
+/****************************************************************************
+ * Name: up_rxflowcontrol
+ *
+ * Description:
+ *   Basic flowcontrol functionality.
+ *   Disable RX interrupts and clear the fifo in case of full RX buffer.
+ *   Enable RX interrupts in case of empty buffer.
+ *   Return true if RX FIFO was cleared.
+ *
+ ****************************************************************************/
+
+static bool up_rxflowcontrol(struct uart_dev_s *dev,
+                                  unsigned int nbuffered, bool upper)
+{
+  struct up_dev_s *priv = (struct up_dev_s *)dev->priv;
+  bool ret = false;
+
+  if (nbuffered == 0 || upper == false)
```

Comment:
This should be enabled when there is room  in the RX buffer, not when it is empty. So, compare  nbuffered against the rx buffer size ?